packages/searchbase/tests/index.test.js

```diff
@@ -207,3 +208,76 @@ it('onSuggestionsQueryChange', async () => {
   searchbase.triggerSuggestionsQuery();
   await new Promise(r => setTimeout(r, 4000));
 });
+
+test('suggestions with subscribeToStateChanges', () => {
+  const searchbase = new Searchbase({
+    index,
+    url,
+    credentials,
+    dataField: 'original_title'
+  });
+  searchbase.subscribeToStateChanges(({ suggestions }) => {
+    expect(suggestions.prev.data).toEqual([]);
+    expect(suggestions.next.data).toEqual([
+      { label: 'hello', value: 'world', source: { _id: 123 } }
+    ]);
+  });
+  searchbase.setSuggestions([
+    { label: 'hello', value: 'world', source: { _id: 123 } }
+  ]);
+});
+
+test('set [value, results, suggestions]', async () => {
+  const searchbase = new Searchbase({
+    index,
+    url,
+    credentials,
+    dataField: 'original_title'
+  });
+  searchbase.onValueChange = (next, prev) => {
+    expect(prev).toEqual('');
+    expect(next).toEqual('hello world');
+  };
+  searchbase.onResults = (next, prev) => {
+    expect(prev.data).toEqual([]);
+    expect(next.data).toEqual([{ key: 'value' }]);
+  };
+  searchbase.onSuggestions = (next, prev) => {
+    expect(prev.data).toEqual([]);
+    expect(next.data).toEqual([
+      { label: 'hello', value: 'world', source: { _id: 123 } }
+    ]);
+  };
+  searchbase.setResults([{ key: 'value' }]);
+  searchbase.setValue('hello world');
+  searchbase.setSuggestions([
```

Comment:
@ShahAnuj2610 I don't think you can test multiple things in a single test, it'll execute the test after the first success.